precompiles/dapps-staking/AstarBase.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+pragma solidity >=0.7.0;
+
+/// Interface to the precompiled contract on Shibuya/Shiden/Astar
+/// Predeployed at the address 0x0000000000000000000000000000000000005001
+interface AstarBase {
+
+    /// @dev Read Staked amount for the staker
+    /// @return Staked amount for the staker
+    function is_in_astarbase(address staker) external view returns (uint128);
```

Comment:
I'm having trouble understanding what the difference is for this function and `read_staked_amount` (https://github.com/AstarNetwork/Astar/pull/517/files#diff-d276a7076f62369a936446902ee693318b51b59bcea0a746a9de84aa5875da13R29) except for the name.

Reply:
there is no difference. I just feel that developer who is using AstarBase does not need to know anything about dApps Staking internals and the name `read_staked_amount` has no meaning to him. Moreover why to use DappsStaking.sol interface when you can have this self-explanatory oneliner  `is_in_astarbase`
There is no further complexity to the code. Both reuse same code in precompile. They just have different selector hex

Reply:
I feel like it is better to make an npm package called `astar-base` that reads the `read_staked_amount` store since this is a utility function for certain devs using existing functionality.
The blockchain should be treated as the optimized source of truth. For utility tools that use existing functions of the chain with a different name should be created as an external utility tool.

Again, this is more related to how we structure and maintain the code even for precompiled, not the functionality (since the idea of AstarBase is really great).